src/aof.c

```diff
@@ -265,7 +417,45 @@ void flushAppendOnlyFile(int force) {
      * or alike */
 
     latencyStartMonitor(latency);
-    nwritten = write(server.aof_fd,server.aof_buf,sdslen(server.aof_buf));
+#ifdef USE_NVML
+    if (server.aof_use_nvml == 1) {
+        size_t len = sdslen(server.aof_buf);
+        size_t newlen = server.aof_current_size + len;
+
+        /* Check if there's required space, if not try to rewrite first */
+        if (newlen > server.aof_nvml_log_size)
```

Comment:
1) IMHO it would be better to simply call *pmemlog_append* and check the status.
2) If there is no space left, then triggering AOF rewrite won't help, as we lost the data anyway.